### PR TITLE
Stop running waypoint as root

### PR DESCRIPTION
**Pull request by howardjohn**
Not sure why we were doing this... seems to work fine without, and we
shouldn't need priv.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml [hunk 1]
```diff
@@ -212,7 +212,10 @@ spec:
         securityContext:
           privileged: false
           runAsGroup: 1337
-          runAsUser: 0
+          runAsUser: 1337
```
Comment (bleggett) [inline manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:215]: Is running as a specific UID/GUID even important for non-sidecar envoys? I wouldn't expect so.

**Review by bleggett — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:215]: No we just need a non-root one really. We already do this for the other gateways so if we should change them lets do it all together?

**Comment by r0bj** [comment]
Related to https://github.com/istio/istio/issues/51291

File: manifests/charts/istio-control/istio-discovery/files/waypoint.yaml [hunk 2]
```diff
@@ -212,7 +212,10 @@ spec:
         securityContext:
```
Comment (r0bj) [inline manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:212]: Consider adding seccompProfile to ensure compliance with the Restricted Pod Security Standards (https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted).
```
          seccompProfile:
            type: RuntimeDefault
```

Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:212]: We haven't used this elsewhere in Istio (and if we do it here, we should be consistent with the rest of Istio) due to the fact `RuntimeDefault` is basically "literally anything the platform decides", which could be somethings more restrictive than what we allow. There are runtimes where the default doesn't even allow pthread_create (??) for instance

Comment (bleggett) [reply manifests/charts/istio-control/istio-discovery/files/waypoint.yaml:212]: If we go down that path, yeah - we might as well just template `seccompProfile`, default it to our current, and allow people to roll the dice with their own seccompProfiles.

**Review by keithmattix — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

**Pull request merged**